packages/counters.lua

```diff
@@ -52,91 +52,103 @@ SILE.formatCounter = function(options)
       display = "roman"
     end
 
-    local ok, res = pcall(function() return icu.format_number(options.value, display) end)
+    local ok, res = pcall(function() return icu.format_number(counter.value, display) end)
     if ok then return res end
   end
 
-  if (options.display == "roman") then return romanize(options.value):lower() end
-  if (options.display == "Roman") then return romanize(options.value) end
-  if (options.display == "alpha") then return alpha(options.value) end
-  return tostring(options.value)
+  if (counter.display == "roman") then return romanize(counter.value):lower() end
+  if (counter.display == "Roman") then return romanize(counter.value) end
+  if (counter.display == "alpha") then return alpha(counter.value) end
+  return tostring(counter.value)
 end
 
-local _init = function (c)
-  if not(SILE.scratch.counters[c]) then
-    SILE.scratch.counters[c] = { value= 0, display= "arabic" }
+local function getCounter(id)
+  local counter = SILE.scratch.counters[id]
+  if not counter then
+    counter = {}
+    SILE.scratch.counters[id] = { value= 0, display= "arabic", format = SILE.formatCounter }
   end
+  return counter
```

Comment:
Why return an empty table in the case of a new counter instead of the actual counter that just got created? The scoping of `counter` in this bit of code is either confusing or the logic is wrong.